src/transpiling/jsx_precompile.rs

```diff
@@ -367,6 +368,124 @@ fn serialize_attr(attr_name: &str, value: &str) -> String {
   format!(" {}=\"{}\"", escape_html(attr_name), escape_html(value))
 }
 
+fn merge_serializable_children(
+  children: &[JSXElementChild],
+) -> (Vec<JSXElementChild>, usize, usize) {
+  // Do a first pass over children to merge sibling text nodes
+  // and check if it contains any serializable nodes.
+  let mut text_count = 0;
+  let mut serializable_count = 0;
+  let mut buf = String::new();
+
+  let mut normalized_children: Vec<JSXElementChild> = vec![];
+  for child in children.iter() {
+    match child {
+      JSXElementChild::JSXText(jsx_text) => {
+        let text = jsx_text_to_str(jsx_text);
+        if text.is_empty() {
+          continue;
+        }
+        text_count += 1;
+
+        buf.push_str(&text);
+      }
+      JSXElementChild::JSXExprContainer(jsx_expr_container) => {
+        match &jsx_expr_container.expr {
+          // Empty JSX expressions can be ignored as they have no content
+          // Case: <div>{}</div>
+          // Case: <div>{/* fooo */}</div>
+          JSXExpr::JSXEmptyExpr(_) => {}
+          // Case: <div>{"foo"}</div>
+          // Case: <div>{2 + 2}</div>
+          JSXExpr::Expr(expr) => {
+            if let Expr::Lit(lit) = *expr.clone() {
```

Comment:
You probably can get away without a clone here (`&*expr`)

Reply:
I tried that, but then it complains with this:

```sh
error[E0308]: mismatched types
   --> src/transpiling/jsx_precompile.rs:401:20
    |
401 |             if let Expr::Lit(lit) = &*expr {
    |                    ^^^^^^^^^^^^^^   ------ this expression has type `&Box<swc_ecma_ast::Expr>`
    |                    |
    |                    expected `Box<Expr>`, found `Expr`
    |
    = note: expected struct `Box<swc_ecma_ast::Expr>`
                 found enum `swc_ecma_ast::Expr`
```

Reply:
Try `if let Expr::Lit(lit) = &**expr`. It's a reference of a `Box`, so you need to dereference it with `*` then debox it with `*`.